src/fpu/fpu_instructions.h

```diff
@@ -102,9 +103,33 @@ static void FPU_FPOP(void){
 static double FROUND(double in){
 	switch(fpu.round){
 	case ROUND_Nearest: return std::nearbyint(in);
-	case ROUND_Down: return (floor(in));
-	case ROUND_Up: return (ceil(in));
-	case ROUND_Chop: [[fallthrough]]; // cast by the caller chops
+	case ROUND_Down: return std::floor(in);
+	case ROUND_Up: return std::ceil(in);
+	case ROUND_Chop: 
+	{
+		// Most x32/x64 builds use fpu_instructions_x86.h.
+		// If we are here on x64, we don't need this fix.
+#if !defined(__x86_64__) && !defined(_M_X64)
+		// This is a fix for rounding to a close integer in
+		// extended precision mode, e.g. 7.999999999999994; an example
+		// can be seen in the Quake options screen size slider.
+		// In this case, what is almost certainly wanted is 8.0,
+		// so return the closer integer within the tolerance instead
+		// of chopping to the lower value.
+		if (fpu.cw & ExtendedPrecisionModeMask) {
+			constexpr double tolerance = 1e-14;
+
+			if (const auto lower = std::floor(in);
+			    (in - lower) < tolerance) {
+				return lower;
+			} else if (const auto upper = std::ceil(in);
+			           (upper - in) < tolerance) {
+				return upper;
+			}
+		}
```

Comment:
*Maybe* this could cause issues with application software that use heavy numerical calculations, but who knows... I guess only testing will tell!

Reply:
Hopefully it should be more accurate for most cases! Like in Quake, there's that small difference when rounding using truncate (or "chop") mode, that what is calculated as 8.0 on x64, on ARM is e.g. 7.999999999999996, but the chop mode puts that down to 7.

Reply:
A disclaimer section in our webpage is in order, btw. Basically we should state our emulator is only for entertainment purposes and for personal use, and we're not responsible for any loss or catastrophes when used for anything else... 😬

Reply:
> A disclaimer section on our webpage is in order, btw. Basically we should state our emulator is only for entertainment purposes and for personal use, and we're not responsible for any loss or catastrophes when used for anything else... 😬

And a sub-disclaimer for those engineers who, for some strange reason, are using Dosbox Staging to engineer bridges.... [/half joking]

Reply:
> A disclaimer section on our webpage is in order, btw. Basically we should state our emulator is only for entertainment purposes and for personal use, and we're not responsible for any loss or catastrophes when used for anything else... 😬

Yeah we can do something like that.

Reply:
> *Maybe* this could cause issues with application software that use heavy numerical calculations, but who knows... I guess only testing will tell!

With rounding pumped up a notch, we should see improvements in truss thicknesses and I-beam width calculations, so perhaps safer buildings and bridges: 👍 👷

I wish I had some of those old posts.. but yeah: there are some very old civil engineering packages that people are still using and teaching with in schools (and using dosbox!)

Reply:
I think there should be a setting where we default to this ON and optionally allowing people to run with more precision if they so desire (on ARM platforms).

Reply:
But that can be implemented as a follow-up PR.

Reply:
This is an awfully big hammer for one feature in one game...

Reply:
Its Quake. 🙂